// @ionic/app-scripts/dist/core/bundle-components.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var logger_1 = require("../logger/logger");
var fs = require("fs");
var path = require("path");
var nodeSass = require("node-sass");
var rollup = require("rollup");
var typescript = require("typescript");
var uglify = require("uglify-js");
function bundleCoreComponents(context) {
    var compiler = getCoreCompiler(context);
    if (!compiler) {
        logger_1.Logger.debug("skipping core component bundling");
        return Promise.resolve();
    }
    var config = {
        srcDir: context.coreDir,
        destDir: context.buildDir,
        packages: {
            fs: fs,
            path: path,
            nodeSass: nodeSass,
            rollup: rollup,
            typescript: typescript,
            uglify: uglify
        }
    };
    return compiler.bundle(config).then(function (results) {
        if (results.errors) {
            results.errors.forEach(function (err) {
                logger_1.Logger.error("compiler.bundle, results: " + err);
            });
        }
    }).catch(function (err) {
        logger_1.Logger.error("compiler.bundle: " + err);
    });
}
exports.bundleCoreComponents = bundleCoreComponents;
function getCoreCompiler(context) {
    try {
        return require(context.coreCompilerFilePath);
    }
    catch (e) {
        logger_1.Logger.debug("error loading core compiler: " + context.coreCompilerFilePath + ", " + e);
    }
    return null;
}
